### DISCUSSION TITLE
Deconstruct delegate parameter

### DISCUSSION THREAD

**Discussion by ziaulhasanhamim** [dotnet/csharplang#6354] [Language Ideas] [answered]
Deconstruct pattern gives the perfect syntax to get precisely what you need from a big object. 

This currently works in c#
```c#
Action<(int, int)> act =  tup => 
{
  var (i, j) = tup;
  // method body
};
```
But this can be simplified 
```c#
Action<(int, int)> act =  ((var i, var j)) => 
{
  // method body
};
// or 
Action<(int, int)> act =  (var (i, j)) => 
{
  // method body
};
```

This can be extended to properties too #6355
```c#
Action<A> act = (A { I: var i}) =>
{
  //method body
};

public class A
{
  public int I { get; set; }
}
```

Note that there will be no runtime pattern matching there. So it can be decided in compile time whether or not the lambda can be cast to the specified delegate

**Comment by svick** [comment, marked as answer]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/258.

**Reply by ziaulhasanhamim** [reply]
Yes I didn't notice that issue. But the property deconstruct can be useful
